pkg/rpc/client/rpc.go

```diff
@@ -1041,6 +1041,12 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 // AddNetworkFee adds network fee for each witness script and optional extra
 // network fee to transaction. `accs` is an array signer's accounts.
 func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, accs ...*wallet.Account) error {
+	if tx == nil {
+		return errors.New("transaction is nil")
+	}
+	if len(tx.Script) == 0 {
```

Comment:
Initially I've though about `tx.IsValid()`, since we shouldn't be bothered with tx details here, but then, how about
```
--- a/pkg/core/transaction/transaction.go
+++ b/pkg/core/transaction/transaction.go
@@ -226,10 +226,6 @@ func (t *Transaction) EncodeBinary(bw *io.BinWriter) {
 // encodeHashableFields encodes the fields that are not used for
 // signing the transaction, which are all fields except the scripts.
 func (t *Transaction) encodeHashableFields(bw *io.BinWriter) {
-       if len(t.Script) == 0 {
-               bw.Err = errors.New("transaction has no script")
-               return
-       }
        bw.WriteB(byte(t.Version))
        bw.WriteU32LE(t.Nonce)
        bw.WriteU64LE(uint64(t.SystemFee))
```
Garbage in -> garbage out. But it won't panic, it can be serialized.

Reply:
Makes sense, ССЗБ, fixed.

Reply:
Do we need the second commit, just in case?

Reply:
I think we can avoid changing `AddNetworkFee` at all. `nil` tx is a pure developer error.

Reply:
Fixed.

Reply:
So why do we change `AddNetworkFee()`?

Reply:
I accidently squashed the last commit instead of dropping it during interactive rebase, fixed.